lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
+      attachment_references.each do |attachment_reference|
+        unless attachment_reference_ids.include?(attachment_reference.id)
+          attachment_reference.mark_for_destruction
+        end
+      end
+    end
+
+    # Find all attachment_reference ids in the columns specified.
+    #
+    # @return [Array<Integer>]
+    def active_attachment_reference_ids
+      attachment_reference_ids = []
+      columns = self.class.attachable_columns || []
+      columns.each do |column|
+        content = send(column)
+        attachment_reference_ids += parse_attachment_reference_ids_from_content(content)
```

Comment:
it might be better to define another accessor, e.g. `column_name_attachment_references` (`content_attachment_references`) which defaults to parsing a string, so that models can override how a column's references is retrieved. include that in the documentation?